### ISSUE TITLE
Botkit only responds to one `hears` match 

### ISSUE THREAD

**Issue by nadavc** [howdyai/botkit#696]
This has been previously reported in #286, but appears to have been closed so I'm opening a new issue. 

Botkit was designed to only respond to the first match on `controller.hears()`. Specifically, whenever a callback is called in (See https://github.com/howdyai/botkit/blob/master/lib/CoreBot.js#L963), it returns `false`, which causes the event handling loop to exit early (https://github.com/howdyai/botkit/blob/master/lib/CoreBot.js#L989). 

This is usually ok for most of our use cases, but sometimes we want to respond to more than one match. As far as I can tell, the only way to achieve this now is to add a `controller.hears()` regexp that specifically looks for that combination.

I wrote a quick script to demonstrate the problem in https://github.com/nadavc/botkit-multi-hears/blob/master/bot.js. To reproduce:

* Replace `xoxb...` with a real Slack token
* Send `@here` to where a channel where the bot is listening on

The result - only the response for `@here` is executed. The other listener (which should match everything) is ignored.

Could you please explain the reasoning for breaking out of the response loop early? Could we just remove/comment out the lines at https://github.com/howdyai/botkit/blob/master/lib/CoreBot.js#L989-L990 ? 

Thanks!

**Comment by ouadie-lahdioui** [comment]
Hello @nadavc, 

And when you do like that :
```
controller.hears(["@here", "ambient"], (bot, message) => {
   console.log("answering @here")   
})
````
What is the result ?

**Comment by tiagosiebler** [comment]
Why was this closed? No plans to address if someone wants separate controller.hears to trigger if they all match? Seems messy to have everything in a single controller.hears @peterswimm

**Comment by peterswimm** [comment]
@tiagosiebler closed because the OP never followed up to an asked question. If you have plans or need for a feature, by all means feel free to create a separate topic for discussion of the feature request.

**Comment by nadavc** [comment]
Hi - sorry for missing the original reply. As I mentioned in the original message, that particular piece of code would register and respond to `@here` and won't respond to anything else that matches (including a catch-all regexp like `.*`). It's demonstrated in the sample I prepared and linked to.

**Comment by jonchurch** [comment]
Put multiple regex in an array for the first argument of hears, should
trigger once if any input matches the regexs.

**Comment by ouadie-lahdioui** [comment]
Exactly, try this :

`````
controller.hears(["@here", "ambient"], (bot, message) => {
   console.log("answering @here")   
})
````

instead of this :

````
controller.hears("@here", ["ambient"], (bot, message) => {
   console.log("answering @here")   
})
````

**Comment by jonchurch** [comment]
Well it would be controller.hears(['^hello$', '^hi there$'], 'ambient',
function(bot,message){})

To trigger both hi and hello there

Also, @here is parsed out Im pretty sure by slack and you wouldnt be able
to trigger it anyways I believe.

**Comment by nadavc** [comment]
Yes, that's right. To simplify my question -- is it possible to specify a trigger for `expr` using one `controller.hears()` call, and then specify another catchall trigger that would also fire when the `expr` is matched.

(btw our bot has been responding to `@here` for quite a while now...)
